exercises/concept/top-secret/.docs/hints.md

```diff
@@ -0,0 +1,52 @@
+# Hints
+
+## General
+
+- Read about quoting in the [official Getting Started guide][getting-started-quote].
+- Read the [introduction to Elixir AST by Lucas San Román][ast-intro-lucas].
+- Read [the official documentation for `quote`][doc-quote].
+- Inspect the output of [`quote`][doc-quote] to familiarize yourself with how ASTs look like for specific code snippets.
+
+## 1. Turn code into data
+
+- There is a [built-in function][doc-code-string-to-quoted] that turns a string with code into an AST.
+
+## 2. Parse a single AST node
+
+- Inspect the output of [`quote`][doc-quote] to familiarize yourself with how ASTs look like for specific code snippets.
+- The operations that define a function are `:def` and `:defp`.
+- The operation is the first element in a three-element AST node tuple.
+- You can ignore the second element in the tuple in this exercise completely.
+- The third element in the tuple is the argument list of the operation that defines the function.
+- The first element on that list is the function's name, and the second element is the function's body.
+- The exact pattern to match an AST node that defines a function and get its name is: `{op, _, [{name, _, _} | _]} when op in [:def, :defp]`.
```

Comment:
Yeah, maybe, I wasn't sure.. I'm a bit worried that some people might not figure it at at all on their own, but you're right. I'll remove it.

The only other exercise where we have a very revealing hint like that is for Captain's Log, for the formatting string for `:iolib.format`, and that's because Erlang documentation for that topic is... written in a way that I don't understand it at all 🙂